File: flink-utils/src/main/scala/io/github/ppdzm/utils/flink/source/kafka/AvroArrayRecordFlinkKafkaStreaming.scala
package io.github.ppdzm.utils.flink.source.kafka

import org.apache.avro.generic.{GenericData, GenericRecord}
import org.apache.flink.api.common.serialization.DeserializationSchema
import org.apache.flink.formats.avro.RegistryAvroDeserializationSchema

/**
 * Created by Stuart Alex on 2021/4/6.
 */
trait AvroArrayRecordFlinkKafkaStreaming extends AvroFlinkKafkaStreaming[GenericData.Array[GenericRecord]] {
    /**
     * [[DeserializationSchema]]
     */
    override protected lazy val deserializationSchema = new RegistryAvroDeserializationSchema(classOf[GenericData.Array[GenericRecord]], schema, schemaCoderProvider)
}

File: flink-utils/src/main/java/io/github/ppdzm/utils/flink/streaming/kafka/AvroArrayRecordFlinkKafkaStreaming.java
package io.github.ppdzm.utils.flink.streaming.kafka;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.apache.avro.generic.GenericData;
import org.apache.avro.generic.GenericRecord;

/**
 * @author Created by Stuart Alex on 2021/5/8.
 */
@Data
@EqualsAndHashCode(callSuper = true)
public abstract class AvroArrayRecordFlinkKafkaStreaming extends AvroFlinkKafkaStreaming<GenericData.Array<GenericRecord>> {
}
